Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py

```diff
@@ -35,152 +37,141 @@ def name(self):
 
     def summary(self):
         """Return a summary of the algorithm."""
-        return "Apply calibration data to detector pixel Y locations"
+        return "Corrects the positions of detector pixels by applying TwoTheta offsets provided in a calibration file"
 
     def seeAlso(self):
         """Return a list of related algorithm names."""
-        return ["ReflectometryISISLoadAndProcess"]
+        return ["ReflectometryISISLoadAndProcess", "SpecularReflectionPositionCorrect"]
 
     def PyInit(self):
         self.declareProperty(
             WorkspaceProperty(self._WORKSPACE, "", direction=Direction.Input, optional=PropertyMode.Mandatory),
-            doc="An input workspace or workspace group",
+            doc="An input workspace or workspace group.",
         )
         self.declareProperty(
             FileProperty(self._CALIBRATION_FILE, "", action=FileAction.OptionalLoad, direction=Direction.Input, extensions=["dat"]),
-            doc="Calibration data file containing Y locations for detector pixels in mm.",
+            doc="Calibration data file containing a list of detector IDs and twoTheta offsets (in degrees)."
+            f"These should be provided as two spaced-delimited columns, labelled {self._DET_ID_LABEL} and {self._THETA_LABEL}.",
         )
         self.declareProperty(
             WorkspaceProperty(self._OUTPUT_WORKSPACE, "", direction=Direction.Output), doc="The calibrated output workspace."
         )
-        self.copyProperties("ReflectometryReductionOneAuto", [self._DEBUG])
 
     def PyExec(self):
-        ws = self.getProperty(self._WORKSPACE).value
-        det_info = ws.detectorInfo()
-        comp_info = ws.componentInfo()
+        # Set the expected order of the columns in the calibration file
+        self._det_id_col_idx = 0
+        self._theta_offset_col_idx = 1
 
-        specular_pixel_idx = self._find_specular_pixel_index(ws, det_info)
-        calib_table = self._create_calibration_table_from_scan(ws, det_info, comp_info, specular_pixel_idx)
+        try:
+            calibration_data = self._parse_calibration_file(self._calibration_filepath)
+        except FileNotFoundError:
+            raise FileNotFoundError("Calibration file path cannot be found")
 
-        # Apply calibration to workspace
-        output_ws = CloneWorkspace(InputWorkspace=ws)
-        ApplyCalibration(Workspace=output_ws, CalibrationTable=calib_table)
-        self.setProperty(self._OUTPUT_WORKSPACE, output_ws)
-        AnalysisDataService.remove("output_ws")
-        if self.getProperty(self._DEBUG).isDefault:
```

Comment:
The debug property was useful originally when we were using the ApplyCalibration algorithm, as we first created a TableWorkspace with all the values to be applied and then passed that into the algorithm. In debug mode we would output that TableWorkspace to the ADS. However now we don't build a table, each pixel is moved individually so there is no debug table to output. If users need to check the new TwoTheta values that have been calculated and applied to each detector they can now see that via the workspace history.

Reply:
Since my latest changes it won't be possible to see the history any more, but users can check the theta values in the detector table if they want to inspect the output of the calibration step. This should give sufficient visibility - the other reason for having the explicit debug option previously was that when we were changing detector positions instead of two theta values it would have been harder for them to see the effect of the calibration.